### ISSUE TITLE
`match_wildcard_for_single_variants` triggers on guarded wild matches

### ISSUE THREAD

**Issue by Jarcho** [rust-lang/rust-clippy#9993]
### Summary

`match_wildcard_for_single_variants` will match on guarded match arms even when it can match multiple variants.

### Reproducer

I tried this code:

```rust
enum Foo {
    A(bool),
    B,
}

#[warn(clippy::pedantic)]
fn main() {
    let _ = match Foo::A(true) {
        _ if false => 0,
        Foo::A(true) => 1,
        Foo::A(false) => 2,
        Foo::B => 3,
    };
}
```

I expected to see this happen: No warning

Instead, this happened:

```
warning: wildcard matches only a single variant and will also match any future added variants
 --> src/main.rs:9:9
  |
9 |         _ if false => 0,
  |         ^ help: try this: `Foo::A(_)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_wildcard_for_single_variants
note: the lint level is defined here
 --> src/main.rs:6:8
  |
6 | #[warn(clippy::pedantic)]
  |        ^^^^^^^^^^^^^^^^
  = note: `#[warn(clippy::match_wildcard_for_single_variants)]` implied by `#[warn(clippy::pedantic)]`
```

### Version

```text
nightly-2022-11-29
```
